x-pack/plugins/fleet/public/applications/integrations/sections/epm/components/integration_preference.tsx

```diff
@@ -86,42 +86,39 @@ const options: Option[] = [
 export const IntegrationPreference = ({
   initialType,
   onChange,
-  onPrereleaseEnabledChange,
+  prereleaseIntegrationsEnabled,
 }: Props) => {
   const [idSelected, setIdSelected] = React.useState<IntegrationPreferenceType>(initialType);
-
-  const { docLinks } = useStartServices();
-
-  const [prereleaseIntegrationsEnabled, setPrereleaseIntegrationsEnabled] = React.useState<
+  const [prereleaseIntegrationsChecked, setPrereleaseIntegrationsChecked] = React.useState<
     boolean | undefined
   >(undefined);
 
-  const { data: settings, error: settingsError } = useGetSettings();
-
-  useEffect(() => {
-    const isEnabled = Boolean(settings?.item.prerelease_integrations_enabled);
-    if (settings?.item) {
-      setPrereleaseIntegrationsEnabled(isEnabled);
-    } else if (settingsError) {
-      setPrereleaseIntegrationsEnabled(false);
-    }
-  }, [settings?.item, settingsError]);
-
-  useEffect(() => {
-    if (prereleaseIntegrationsEnabled !== undefined) {
-      onPrereleaseEnabledChange(prereleaseIntegrationsEnabled);
-    }
-  }, [onPrereleaseEnabledChange, prereleaseIntegrationsEnabled]);
-
-  const updateSettings = useCallback(async (prerelease: boolean) => {
-    const res = await sendPutSettings({
-      prerelease_integrations_enabled: prerelease,
-    });
-
-    if (res.error) {
-      throw res.error;
-    }
-  }, []);
+  const { docLinks, notifications } = useStartServices();
+
+  const { mutateAsync: mutateSettingsAsync } = usePutSettingsMutation();
+
+  const updateSettings = useCallback(
+    async (prerelease: boolean) => {
+      try {
+        setPrereleaseIntegrationsChecked(prerelease);
+        const res = await mutateSettingsAsync({
+          prerelease_integrations_enabled: prerelease,
+        });
+
+        if (res.error) {
+          throw res.error;
+        }
+      } catch (error) {
+        setPrereleaseIntegrationsChecked(!prerelease);
+        notifications.toasts.addError(error, {
```

Comment:
Added some error handling, as user without the put settings permissions cannot change the prerelease toggle.